int8 xSide
int8 ySide
float64 xDis
float64 yDis
